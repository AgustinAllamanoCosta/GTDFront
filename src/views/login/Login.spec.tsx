import React from 'react';
import { expect } from '@jest/globals';
import { render } from '@testing-library/react';
import LoginView from './Login';
import GoogleAuthContext from '../../components/useGoogleAuth/useGoogleAuth';
import { BrowserRouter } from 'react-router-dom';

describe('Login View', () => {
  test('Should render a login view with singin and login buttons', async () => {
    const { container } = render(
      <BrowserRouter>
        <GoogleAuthContext clientId="CLIENT-ID">
          <LoginView />
        </GoogleAuthContext>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
