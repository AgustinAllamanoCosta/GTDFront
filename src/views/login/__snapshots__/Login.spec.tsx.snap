// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login View Should render a login view with singin and login buttons 1`] = `
<div>
  <div
    class="sc-kPNBzt hWCVUV"
  >
    <div
      class="sc-bXxmBy fAEwdg"
    >
      <div
        class="sc-dBwcjI fsysFb"
        data-cy="Card-Contents"
        is_center="true"
        padding="true"
      >
        <a
          class="sc-dMnqAs dOugPz"
          data-cy="Login-Title"
          href="https://github.com/AgustinAllamanoCosta/GTDFront"
          target="_blank"
        >
          Get Things Done
        </a>
        <div
          class="sc-jlOLRj jnZxFa"
        >
          <div
            class="sc-hfDuwB hZBRcw"
            data-cy="button-Login With Google"
          >
            <span
              class="sc-jCIOXC cfzUow"
              data-cy="button-text"
            >
              Login With Google
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
